Common/Net/NetBuffer.cpp

```diff
@@ -37,8 +37,9 @@ bool Buffer::FlushSocket(uintptr_t sock, double timeout, bool *cancelled) {
 			}
 		}
 		int sent = send(sock, &data_[pos], (int)(end - pos), MSG_NOSIGNAL);
+		// TODO: Do we need some retry logic here, instead of just giving up?
```

Comment:
I suppose we should try again, but only for certain error codes.  In theory EINTR/EAGAIN/etc. would mean a retry, but most other errors that occur wouldn't.

-[Unknown]

Reply:
The one I was seeing while running under Valgrind was EPIPE. Not sure if valgrind is intentionally causing it to test robustness, or what's going on with that. Because outside of valgrind it doesn't hit this.